storage/src/test/java/tech/pegasys/teku/storage/server/AbstractDatabaseTest.java

```diff
@@ -490,6 +496,53 @@ public void testShouldRecordFinalizedBlocksAndStatesInBatchUpdate()
     testShouldRecordFinalizedBlocksAndStates(StateStorageMode.ARCHIVE, true);
   }
 
+  @Test
+  public void shouldRecreateMerkleTreeFromDatabase() {
+    final DataStructureUtil dataStructureUtil = new DataStructureUtil();
+    DepositWithIndex firstDeposit = dataStructureUtil.randomDepositWithIndex(1L);
+    DepositWithIndex secondDeposit = dataStructureUtil.randomDepositWithIndex(2L);
+    eth1DepositChannel.addEth1Deposit(firstDeposit);
+    eth1DepositChannel.addEth1Deposit(secondDeposit);
+
+    assertThat(eth1DepositChannel.getDeposits().get(0)).isEqualTo(firstDeposit);
+    assertThat(eth1DepositChannel.getDeposits().get(1)).isEqualTo(secondDeposit);
```

Comment:
Why are we asserting that our stub works here?  Seems like we should just call `database.addEth1Deposit` directly and remove `TrackingEth1DepositChannel`.

Reply:
i started checking that everywhere but it didn't seem to help at all once i knew the database was working. I built it out on the basis the db wouldn't work straight away so that i could debug. i'll clean it up.